### DISCUSSION TITLE
[Documentation/Wiki] FAQ entry about frozen strings?

### DISCUSSION THREAD

**Discussion by rubyFeedback** [jruby/jruby#6626] [Q&A] [answered]
Hey there headius and others,

I have used ruby for many years; slowly learning ruby and getting
into jruby too. Even using swing stuff for GUI!

Anyway - there is a super-simple question I have.

Does:

"# frozen_string_literal: true"

In a given .rb file, have any effect via jruby?

Right now I am invoke it on linux in this way:

    jruby foobar.rb

And that works, the swing-JButton works fine, all cool. I don't
know if that has any effect, though.

Would it be possible to add an entry to the FAQ about it?
E. g. just something like:

QUESTION: Does using "# frozen_string_literal: true" in a .rb file have any performance-specific effect?
ANSWER: Yes/No # I really don't know the answer right now which is why I asked here

Reason why this would be nice to the FAQ would be so that it is not forgotten; github issues tend to
be a bit volatile when it comes to knowledge base (people will forget old issues and discussions there
which is why I think the FAQ at the wiki would be nice, like mediawiki style).

**Comment by headius** [comment, marked as answer]
The frozen string directive will have the same effect on performance as it would in standard Ruby: less allocation of String objects.

Ruby Strings passed to Java calls are always converted into Java Strings, which are themselves immutable, so this doesn't really change anything about calling into Java. It may offer a possible performance improvement in the future if we cache the Java String object that corresponds to a given frozen literal String (but we are reluctant to fill memory with more String data than necessary).
